### PR TITLE
Use zizmor to lint GitHub Actions

### PR DESCRIPTION
**Pull request by santisoler**
Make use of `zizmor` to lint GitHub Actions workflows and catch potential security issues. Add `zizmor` to the `environment.yml` and to the `requirements-style.txt`. Add a new `check-actions` target in the `Makefile` that runs `zizmor` on every workflow. Add a new workflow for running `zizmor` on PRs and on pushes to `main`. Apply some minor changes to the script that deploys docs to GitHub Pages after suggestions obtained by running `shellcheck` on it.

**Relevant issues/PRs:**

Part of fatiando/community#159

### CODE REVIEW

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -151,38 +155,41 @@ jobs:
...
           # Need to have this file so that Github doesn't try to run Jekyll
           touch .nojekyll
 
           # Delete all the files and replace with our new  set
           echo -e "\nRemoving old files from previous builds of ${version}:"
-          rm -rvf ${version}
+          rm -rvf "${version}"
           echo -e "\nCopying HTML files to ${version}:"
-          cp -Rvf ../doc/_build/html/ ${version}/
+          cp -Rvf ../doc/_build/html/ "${version}"/
```
Comment (santisoler) [inline .github/workflows/docs.yml:186]:
```suggestion
          cp -Rvf ../doc/_build/html/ "${version}/"
```

File: .github/workflows/docs.yml [hunk 2]
```diff
@@ -151,38 +155,41 @@ jobs:
...
+            version=dev
           fi
           echo "Deploying version: $version"
 
           # Make the new commit message. Needs to happen before cd into deploy
           # to get the right commit hash.
           message="Deploy $version from $(git rev-parse --short HEAD)"
 
-          cd deploy
+          cd deploy || exit
```
Comment (santisoler) [inline .github/workflows/docs.yml:177]:
```suggestion
          cd deploy || exit 1
```

File: .github/workflows/actions.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+          python-version: "3.12"
+
+      - name: Install requirements
+        run: python -m pip install -r env/requirements-style.txt
+
+      - name: List installed packages
+        run: python -m pip freeze
+
+      - name: Lint GitHub Actions
+        run: make check-actions
```
Comment (santisoler) [inline .github/workflows/actions.yml:35]: @leouieda, I'm considering adding the GITHUB_TOKEN here to allow zizmor to check for vulnerabilities online (using the GitHub API). I think this is safe, but wanted to double check with you before applying this change.

```suggestion
      - name: Lint GitHub Actions
        run: make check-actions
        env:
          # Set GH_TOKEN to allow zizmor to check online vulnerabilities
          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

My reasoning: the `GITHUB_TOKEN` [is passed to runners triggered from a forked repository](https://docs.github.com/en/actions/security-for-github-actions/security-guides/using-secrets-in-github-actions#using-secrets-in-a-workflow), but it has [read-only permissions](https://docs.github.com/en/actions/security-for-github-actions/security-guides/automatic-token-authentication#permissions-for-the-github_token) is such case. So no PR from external fork can gain writing permissions to our repo  by exploiting that token.

Comment (leouieda) [reply .github/workflows/actions.yml:35]: Sounds good. The matter is trusting zizmor to not steal tokens, right?

Comment (santisoler) [reply .github/workflows/actions.yml:35]: Yes. I think that's safe within our thread model.

**Pull request merged**